pkg/kubelet/image_manager_test.go

```diff
@@ -32,14 +32,17 @@ import (
 
 var zero time.Time
 
-func newRealImageManager(t *testing.T, policy ImageGCPolicy) (*realImageManager, *dockertools.FakeDockerClient, *cadvisor.Mock) {
+func newRealImageManager(policy ImageGCPolicy) (*realImageManager, *dockertools.FakeDockerClient, *cadvisor.Mock) {
 	fakeDocker := &dockertools.FakeDockerClient{
 		RemovedImages: util.NewStringSet(),
 	}
 	mockCadvisor := new(cadvisor.Mock)
-	im, err := newImageManager(fakeDocker, mockCadvisor, policy)
```

Comment:
Yes, what we have today leaves little reason to separate the garbage collection since we do it right after the new.

We should probably move most of this work to Run(). Today we do it in New(), after New(), and in Run(). The only reason to not do it in New() is when RunOnce() is called, I don't think we want any of those to be running in that case anyways. I'll file an issue and send some PRs tomorrow :)